### DISCUSSION TITLE
Binding to a Dropdown

### DISCUSSION THREAD

**Discussion by gallwood** [BlazorFluentUI/BlazorFluentUI#360] [Q&A] [answered]
Hi,

Does anyone have any advice how to use the SelectedOptionExpression? I have a Model object that I am currently using to bind fields to say TextFields. I also have a property that I wanted to bind to a Dropdown component, but the SelectedOption property is of type IDropdownOption but in my Model, I want to bind to a field that represents a key from the DropdownOptions I supply to the Dropdown.

I was wondering how other people bind to this component. I was hoping I didn't have to write any code, i.e. a handler for the SelectedOptionChange event.

Thanks

Graham

**Comment by limefrogyank** [comment, marked as answer]
`SelectedOptionExpression` is only for DataValidation, I think.  When you set the expression (it's basically the same as creating a plain `Func`), a `FieldIdentifier` gets created which ties into the `EditForm` you'll have on your page.  Beyond that, I don't remember much about it.

When I made Dropdown, I remember debating over requiring use of a key field vs just forcing people to use an interface.  I don't remember why, but the interface seemed the lesser of the two evils.

Use Linq and just tranform your data into the right interface.  When you get the IDropdown selected object back, you could just put some logic in the setter to parse out the field you want...
